cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3097,6 +3159,7 @@ private boolean isCustomCertUpToDate(StatefulSet sts, Pod pod, String annotation
          */
         private String getReasonsToRestartPod(StatefulSet sts, Pod pod,
                                        boolean nodeCertsChange,
+                                       Boolean kafkaPodUpdatedDynamically,
```

Comment:
Given that `kafkaPodsUpdatedDynamically` is a member wouldn't it be better to just let this method access it, rather than expecting the caller to get from the map?

Reply:
I think it is getting this more complicated because we need to decide there whether it is zk or Kafka pod.